src/utils/rules.ts

```diff
@@ -250,12 +296,18 @@ const getSpaceValue = scale => propVal => {
 // @ts-ignore FIXME
 export const space = props => {
   // test for spacing props, so m* and p*
-  const keys = Object.keys(props).filter(key => /^[mp][trblxy]?$/.test(key));
   // get space configuration from theme
   const scale = get(props.theme, 'base.space');
   // function to convert propVal -> cssVal
   const getStyle = getSpaceValue(scale);
 
+  // to ensure proper order
```

Comment:
Would be nice to extract the keys and sort them somehow, but looking at the code we may need to supply the order list manually.

```js
const order = ['m', 'p', 'x', 'y', 't', 'r', 'b', 'l'];
const keys = Object.keys(props)
  .filter(key => /^[mp][trblxy]?$/.test(key))
  .sort((keyA, keyB) => order.indexOf(keyA[keyA.length - 1]) - order.indexOf(keyB[keyB.length - 1]));
```
It doesn't look super clean, though, so I leave it to you.

Reply:
Leaving as is for readability can go back later to restructure